x-pack/plugins/maps/public/classes/styles/vector/components/legend/extract_color_from_style_property.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { getColorRampCenterColor, getColorPalette } from '../../../color_utils';
+import { COLOR_MAP_TYPE, STYLE_TYPE } from '../../../../../../common/constants';
+import {
+  ColorDynamicOptions,
+  ColorStylePropertyDescriptor,
+} from '../../../../../../common/descriptor_types';
+
+export function extractColorFromStyleProperty(
+  colorStyleProperty: ColorStylePropertyDescriptor | undefined,
+  defaultColor: string
+): string {
+  if (!colorStyleProperty) {
+    return defaultColor;
+  }
+
+  if (colorStyleProperty.type === STYLE_TYPE.STATIC) {
+    return colorStyleProperty.options.color;
+  }
+
+  const dynamicOptions: ColorDynamicOptions = colorStyleProperty.options;
+
+  // Do not use dynamic color unless configuration is complete
+  if (!dynamicOptions.field || !dynamicOptions.field.name) {
+    return defaultColor;
+  }
+
+  if (dynamicOptions.type === COLOR_MAP_TYPE.CATEGORICAL) {
+    if (dynamicOptions.useCustomColorPalette) {
+      return dynamicOptions.customColorPalette && dynamicOptions.customColorPalette.length
+        ? dynamicOptions.customColorPalette[0].color
```

Comment:
The git diffs are not very good, but this is the line that fixed the problem. Added typing so errors like this would be caught. Change file to TS flagged `customColorPalette[0].colorCategory`